### DISCUSSION TITLE
Is Caching of type List<Map<String, Object>  thread safe?

### DISCUSSION THREAD

**Discussion by jbleeok** [ben-manes/caffeine#914] [Q&A] [answered]
If have a cache definition as below. Will it thread safe? It looks working.
static Cache<String, List<Map<String, Object> cacheList = Caffeine.newBuilder().build();
TIA

**Comment by ben-manes** [comment, marked as answer]
Yes, this cache library is thread safe

**Reply by jbleeok** [reply]
Great!!!
Thank you for the answer.

**Reply by ben-manes** [reply]
Node that if you need to do complex operations on the value, that may not be thread-safe. So for example `cacheList.get(key).add(Map.of())` would make a thread-safe call to the cache, but mutating the `List` may not be (e.g. if an ArrayList). Typically one treats a cache's value as immutable. If you do need to do complex operations, then you might use an `asMap()` compute method or a copy-on-write data structure (e.g. `CopyOnWriteArrayList`). This library is thread-safe but your usages may not be, so be thoughtful about concurrent reads when mutating a value.
